### PR TITLE
[SPARK-29084][SQL][TESTS] Check method bytecode size in BenchmarkQueryTest

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This pr proposes to check method bytecode size in `BenchmarkQueryTest`. This metric is critical for performance numbers.

### Why are the changes needed?

For performance checks

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by maropu** [comment]
I found the test below in `TPCDSQuerySuite` fails:
```
- modified-q3 *** FAILED ***
  12184 was not less than 8000 Found too long generated codes in the WholeStageCodegenExec subtree (id=108685) and JIT optimization might not work: the bytecode size is above the default limit of the OpenJDK HotSpot JVM:
  *(1) Project [ss_sold_date_sk#340, ss_item_sk#342, ss_net_profit#362]
  +- *(1) Filter ((((((((((((ss_sold_date_sk#340 >= 2415355) AND (ss_sold_date_s
```

**Comment by dongjoon-hyun** [comment]
Could you add `[TESTS]` into the PR title, @maropu ?

**Comment by dongjoon-hyun** [comment]
Hi, @maropu . Thank you for adding this. This looks useful.
If `modified-q3` still fails, could you add `[WIP]` on the title to prevent accidental merging this PR?

**Comment by dongjoon-hyun** [comment]
Oh, is this ready, @maropu ?

**Comment by maropu** [comment]
I'm checking if all the tests passed. I persnally think we need more time for #25827 but that pr does not block this pr.

**Comment by maropu** [comment]
retest this please

**Comment by maropu** [comment]
ok, could you check this? @dongjoon-hyun @viirya

**Comment by dongjoon-hyun** [comment]
Of course, @maropu ~ I'll do today.

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSQuerySuite.scala [hunk 1]
```diff
@@ -82,13 +82,17 @@ class TPCDSQuerySuite extends BenchmarkQueryTest with TPCDSSchema {
     "q3", "q7", "q10", "q19", "q27", "q34", "q42", "q43", "q46", "q52", "q53", "q55", "q59",
     "q63", "q65", "q68", "q73", "q79", "q89", "q98", "ss_max")
 
+  // List up the known queries having too large code in a generated function
+  val blackListForMethodCodeSizeCheck = Set("modified-q3")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSQuerySuite.scala]: Can we have an IDed TODO to remove this, @maropu ? [SPARK-29128](#25827) is the one right?

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSQuerySuite.scala]: Otherwise, we had better wait for the fix.

Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSQuerySuite.scala]: ok

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @maropu .
Although there is a blacklist query, this PR helps us a lot.
Merged to master.

**Comment by maropu** [comment]
Thanks, @dongjoon-hyun !

**Pull request closed without merging**